src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java

```diff
@@ -194,4 +194,105 @@ public void verifyDataIsTransferredToTargetGCSBucket() {
+    if (result.isPresent()) {
+      pathFromBQTable = result.get();
+    }
+    BeforeActions.scenario.write("GCS bucket filename in BQ Table :" + pathFromBQTable);
+    String filename[] = PluginPropertyUtils.pluginProp(path).split("/");
+    Assert.assertEquals("BQ table output field should contain GCS bucket filename ",
+                        filename[filename.length - 1], pathFromBQTable);
+  }
+
+  @Then("Enter GCS File system properties field {string}")
+  public void enterGCSFileSystemPropertiesField(String property) {
+    CdfGcsActions.enterFileSystemProperties(PluginPropertyUtils.pluginProp(property));
+  }
+
+  @Then("Click on Tidy in GCS File system properties")
+  public void clickOnTidyInGCSFileSystemProperties() {
+    CdfGcsActions.clickFileSystemPropertiesTidyButton();
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with file format {string}")
+  public void verifyDataIsTransferredToTargetGCSBucketWithFileFormat(String fileFormat) {
+    String gcsBucket = TestSetupHooks.gcsTargetBucketName;
+    if (verifyFilePresentInGcsBucket(gcsBucket, "." + fileFormat, "suffix")) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with format "
+                                     + fileFormat + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket + " with format " + fileFormat);
+    }
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with fileName prefix {string}")
+  public void verifyDataIsTransferredToTargetGCSBucketWithFileNamePrefix(String fileNamePrefix) {
+    String gcsBucket = TestSetupHooks.gcsTargetBucketName;
+    fileNamePrefix = PluginPropertyUtils.pluginProp(fileNamePrefix);
+    if (verifyFilePresentInGcsBucket(gcsBucket, fileNamePrefix, "prefix")) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with fileNamePrefix "
+                                     + fileNamePrefix + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket +
+                    " with fileNamePrefix " + fileNamePrefix);
+    }
+  }
+
+  @Then("Verify data is transferred to target GCS bucket with path suffix {string}")
+  public void verifyDataIsTransferredToTargetGCSBucketWithPathSuffix(String pathSuffix) {
+    String gcsBucket = TestSetupHooks.gcsTargetBucketName;
+    pathSuffix = PluginPropertyUtils.pluginProp(pathSuffix);
+    if (verifyFilePresentInGcsBucket(gcsBucket, pathSuffix, "suffix")) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with pathSuffix "
+                                     + pathSuffix + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket +
+                    " with pathSuffix " + pathSuffix);
+    }
+  }
+
+  private boolean verifyFilePresentInGcsBucket(String gcsBucket, String filename, String filePart) {
```

Comment:
make filePart an enum so only specific values are allowed

Reply:
@rmstar filePart Enum created

Reply:
Type of filePart param should be FilePart, not string

Reply:
Changed String param to FilePart